### PR TITLE
mapping from socio2e to socio3e

### PR DESCRIPTION
**Pull request by jarosik10**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1595

### CODE REVIEW

File: src/canonicalBookMap/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Steps to create a map of book modules:
+1. Copy the data from mapping spreadsheet
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/README.md:2]: It would be good to add info what columns do we expect

File: src/canonicalBookMap/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+2. Paste the copied data into template literal
+    const data = `PASTED DATA`
+
+3. Convert the string into array
+    const mappingList = str.split('\n').map((line) => line.split('\t'))
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/README.md:8]: `str` -> `data`

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]

**Comment by PiotrKozlowski** [comment]
I didn't saw it earlier but it seems like 1e should also point to 3e

File: src/canonicalBookMap/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+2. Paste the copied data into template literal
+    const data = `PASTED DATA`
+
+3. Convert the string into array
+    const mappingList = data.split('\n').map((line) => line.split('\t'))
+
+4. Create the map of modules as a string (Modify the moduleMapComment if necessay)
+    const modulesMapStr = mappingList.reduce((prev, current) => {
+        const moduleMapComment = current[0] === current[2] ? `${current[0]} to the same module in 3e` : `${current[0]} to the ${current[2]}`
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/README.md:12]: According to our conversation with Tom: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1595#issuecomment-848812612
it seems like these names not always will be accurate so the single name of the module should be enough

File: src/canonicalBookMap/sociology.ts [hunk 4]
```diff
@@ -7,197 +7,393 @@ export default {
   /* Introduction to Sociology 1e */ 'afe4332a-c97f-4fc4-be27-4e4d384a32d8': [
     /* Introduction to Sociology 2e */ ['02040312-72c8-441e-a685-20e9333f3e1d', {
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/sociology.ts:8]: This should be `3e` and id of `3e` version

Comment (PiotrKozlowski) [reply src/canonicalBookMap/sociology.ts:8]: We should probably also fix comments to contain only names of the old modules

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]

File: src/canonicalBookMap/sociology.ts (hunk 4 above)
Comment (jarosik10) [reply src/canonicalBookMap/sociology.ts:8]: Do we want to change the comments for entire sociology file or the other as well?

Comment (PiotrKozlowski) [reply src/canonicalBookMap/sociology.ts:8]: I would change comments only for this file

File: src/canonicalBookMap/sociology.ts [hunk 5]
```diff
@@ -5,199 +5,395 @@ import { CanonicalBookMap } from '../canonicalBookMap';
...
+      /* Theoretical Perspectives on Culture */
+      '14fba504-dd54-46fb-ad08-9d6b86954ee7': 'ca2347f6-ca5a-46cf-bd8b-75768bd084a3',
+      /* Introduction to Society and Social Interaction */
+      'e9e10f3d-f915-4adf-b0c3-b81080e8bf40': 'ef238876-14ce-464f-ac5a-e30d656d77b7',
+      /* Types of Societies */
+      'cb87c883-2f5c-4d73-ab29-4d2354e2e48e': '526f33fb-2ed8-4d15-b2e5-2b83a2b97849',
+      /* Theoretical Perspectives on Society */
+      'c5b9353a-8e1f-484b-94d4-0d3b886e1661': 'c1bc1aa5-03ef-4909-90b3-f47d33bc0c85',
+      /* Social Constructions of Reality */
       '00b50039-5567-417c-b71b-0df7ef800ebf': 'd06ad985-cecb-48b4-b222-197d04abc6a1',
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/sociology.ts:46]: missed page

File: src/canonicalBookMap/sociology.ts [hunk 6]
```diff
@@ -5,199 +5,395 @@ import { CanonicalBookMap } from '../canonicalBookMap';
...
+      /* 3.4 Theoretical Perspectives on Culture */
+      'f2055c29-6b50-4a77-89d8-f88745ce0f18': 'ca2347f6-ca5a-46cf-bd8b-75768bd084a3',
+      /* 4.0 Introduction to Society and Social interaction */
+      '31eae3d7-1b57-4c24-ad0b-d08a2851b70d': 'ef238876-14ce-464f-ac5a-e30d656d77b7',
+      /* 4.1 Types of Societies */
+      'aace8da1-b83e-4a5e-bfb8-5bd44699381a': '526f33fb-2ed8-4d15-b2e5-2b83a2b97849',
+      /* 4.2 Theoretical Perspectives on Society */
+      '6414e0c9-1e95-4b06-af38-b11762cb667e': 'c1bc1aa5-03ef-4909-90b3-f47d33bc0c85',
+      /* 4.3 Social Constructions of Reality */
+      'd06ad985-cecb-48b4-b222-197d04abc6a': '040ac094-d7eb-4761-840f-04f22aa30797',
```
Comment (PiotrKozlowski) [inline src/canonicalBookMap/sociology.ts:242]: it should be `d06ad985-cecb-48b4-b222-197d04abc6a1`
there is a typo in the spreadsheet - missing `1` at the end and canonical for https://rex-web-issue-1595-vrsz9ftxs2e.herokuapp.com/books/introduction-sociology-2e/pages/4-3-social-constructions-of-reality is not being set

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]
https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1595#issuecomment-849615696

**Review by TomWoodward — APPROVED** [review]
i'm not checking the individual ids, but this looks good otherwise

**Comment by TomWoodward** [comment]
waiting for bey to respond on the issue

**Comment by Bkblodget** [comment]
Mapping doc has been updated with composite pages

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]
https://github.com/openstax/rex-web/pull/1194#issuecomment-853963030

**Review by PiotrKozlowski — APPROVED** [review]

**Comment by PiotrKozlowski** [comment]
@Bkblodget we should probably update also mapping for composite pages from 1e to 3e

**Pull request merged**